query/method/broadcast.json

```diff
@@ -15,10 +15,7 @@
 				"channel": { "description": "[String] name of the channel", "$ref": "channel/subChannel.json" },
 				"message": { 
 					"description": "[Message] message to be published", 
-					"oneOf": [
-						{ "$ref": "message/messageGeneral.json" },
-						{ "$ref": "message/messageWitnessMessage.json" }
```

Comment:
If you delete this, witnesses will not be able to send witness messages.

Reply:
I didn't look into this, but I seem to remember having the same desire to change this.
Are you sure messageWitnessMessage isn't just a subset of messageGeneral? 
You might have answered this to me before, but IIRC messageWitnessMessage is just a messageGeneral with 0 items in the witness_signatures array, which is still a valid messageGeneral

Reply:
I might have missed the difference between to messageGeneral and messageWitnessMessage.

But from what I read, messageWitnessMessage is the same as messageGeneral but it forces witness_signatures to be empty.

That if a message has empty witness_signatures, it fulfill both messageGeneral and messageWitnessMessage. So the oneOf breaks

Reply:
Oh yes @RaulinN , thanks for the reminder.
That constraint is something you can enforce at the protocol specification level (as it was so far), or at the application level ( as some of the other checks are ) .
Either are valid options with a simple tradeoff: the former makes the protocol much more verbose and the application simpler, the latter makes for a neat, concise protocol but it requires additional logic in the application.

Reply:
I also had the feeling that messageWitnessMessage is just a subset of messageGeneral.

If for some reason, it is needed to keep both, json schema also offers anyOf we could use instead of anyOf so that the oneOf doesn't "break"